subgraphs/euler-finance/src/mappings/helpers.ts

```diff
@@ -162,316 +139,607 @@ export function createRepay(event: Repay): BigDecimal {
+    underlyingToken.lastPriceUSD!,
+  );
 
   withdraw.save();
-  market.save();
 
   return withdraw.amountUSD;
 }
 
 export function createLiquidation(event: Liquidation): BigDecimal {
   const liquidation = getOrCreateLiquidate(event);
-  const collateralTokenId = event.params.underlying.toHexString();
+  const underlyingTokenId = event.params.underlying.toHexString();
   const seizedTokenId = event.params.collateral.toHexString();
 
+  const underlyingToken = getOrCreateToken(event.params.underlying);
+  const seizedToken = getOrCreateToken(event.params.collateral);
+
   // repay token market
-  const market = getOrCreateMarket(collateralTokenId);
+  const underlyingAssetStatus = getOrCreateAssetStatus(underlyingTokenId);
+  const collateralAssetStatus = getOrCreateAssetStatus(seizedTokenId);
+  const market = getOrCreateMarket(underlyingAssetStatus.eToken!);
+  const collateralMarket = getOrCreateMarket(collateralAssetStatus.eToken!);
 
-  liquidation.market = collateralTokenId;
-  liquidation.asset = seizedTokenId;
+  liquidation.market = collateralMarket.id;
+  liquidation.asset = underlyingTokenId;
   liquidation.from = event.params.liquidator.toHexString();
-  liquidation.to = market.outputToken!; // eToken transfered to liquidator
+  liquidation.to = market.id; // Market that tokens are repaid to
   liquidation.liquidatee = event.params.violator.toHexString();
-  liquidation.amount = event.params.repay; // Amount is denominated in underlying (not in dToken)
-
-  const collateralMarketUtility = getOrCreateMarketUtility(collateralTokenId);
-  liquidation.amountUSD = amountToUsd(
-    event.params.repay,
-    collateralMarketUtility.twap,
-    collateralMarketUtility.twapPrice,
+  // Amount of collateral liquidated in native units (schema definition)
+  // Amount is denominated in collateral
+  liquidation.amount = bigIntChangeDecimals(event.params._yield, DEFAULT_DECIMALS, seizedToken.decimals);
+  liquidation.amountUSD = bigIntToBDUseDecimals(liquidation.amount, seizedToken.decimals).times(
+    seizedToken.lastPriceUSD!,
   );
-  const collateralMarket = getOrCreateMarket(collateralTokenId);
-  collateralMarket.cumulativeLiquidateUSD = collateralMarket.cumulativeLiquidateUSD.plus(liquidation.amountUSD);
 
-  const seizedMarketUtility = getOrCreateMarketUtility(seizedTokenId);
-  const yieldUSD = amountToUsd(event.params._yield, seizedMarketUtility.twap, seizedMarketUtility.twapPrice);
-  liquidation.profitUSD = yieldUSD.minus(liquidation.amountUSD);
+  const repayUSD = bigIntToBDUseDecimals(event.params.repay, underlyingToken.decimals).times(
+    underlyingToken.lastPriceUSD!,
+  );
+  liquidation.profitUSD = liquidation.amountUSD.minus(repayUSD);
+  liquidation.save();
 
-  if (liquidation.amountUSD) {
-    market.cumulativeLiquidateUSD = market.cumulativeLiquidateUSD.plus(liquidation.amountUSD);
-    market.save();
+  log.info("[createLiquidation]seizedToken={}/{},amount={},price={},amountUSD={},blk={},tx={}", [
```

Comment:
seems like another dev log that we could remove

Reply:
deleted